app/src/main/java/comcmput301f16t01/github/carrier/Requests/RiderRequestActivity.java

```diff
@@ -235,14 +266,32 @@ public void setViews() {
         // TODO why does it say "sarah" even though there is no confirmedDriver?
         // Set up the UsernameTextView of the driver
         UsernameTextView driverUsernameTextView = (UsernameTextView) findViewById(R.id.UsernameTextView_driver);
+        // If no driver has been selected we need to display the list of drivers who have made an offer.
+        TextView driverTextView = (TextView) findViewById(R.id.textView_driver);
+        driverTextView.setText("");
         if (request.getChosenDriver() != null) {
             driverUsernameTextView.setText(request.getChosenDriver().getUsername());
             driverUsernameTextView.setUser(request.getChosenDriver());
+        } else if (request.getOfferedDrivers().size() > 0 && (request.getStatus() != Request.CONFIRMED || request.getStatus() != Request.PAID)) {
+            // If there is an offered driver
+            Toast.makeText(RiderRequestActivity.this, "Here", Toast.LENGTH_SHORT).show();
+            driverTextView = (TextView) findViewById(R.id.textView_driver);
+            driverTextView.setText(R.string.View_Offering_Drivers);
+            driverTextView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    Bundle bundle = getIntent().getExtras();
+                    Intent intent = new Intent(RiderRequestActivity.this, RiderViewOfferingDriversActivity.class);
+                    intent.putExtras(bundle);
+                    startActivity(intent);
+                }
+            });
+
         }
 
         TextView startAddressTextView = (TextView) findViewById(R.id.textView_start);
         String startAddress = request.getStart().getAddress();
-        if(startAddress != null) {
+        if (startAddress != null) {
             startAddressTextView.setText(startAddress);
         } else {
             String startPoint = request.getStart().getLatLong();
```

Comment:
Make this change please. (I know you didn't write these lines but it can be fixed simplified here).